lib/Controller/UserApiController.php

```diff
@@ -44,24 +44,25 @@ public function index(int $documentId, int $sessionId, string $sessionToken, str
 		foreach ($sessions as $session) {
 			$sessionUserId = $session['userId'];
 			if ($sessionUserId !== null && !isset($users[$sessionUserId])) {
-				$users[$sessionUserId] = $this->userManager->getDisplayName($sessionUserId);
+				$displayName = $this->userManager->getDisplayName($sessionUserId);
+				if (stripos($displayName, $filter) !== false) {
+					$users[$sessionUserId] = $displayName;
+				}
 			}
 		}
 
 		$currentSession = $this->sessionService->getSession($documentId, $sessionId, $sessionToken);
 		if ($currentSession->getUserId() !== null) {
 			// Add other users to the autocomplete list
 			[$result] = $this->collaboratorSearch->search($filter, [IShare::TYPE_USER], false, $limit, 0);
+			$userSearch = array_merge($result['users'], $result['exact']['users']);
 
-			foreach ($result['users'] as ['label' => $label, 'value' => $value]) {
+			foreach ($userSearch as ['label' => $label, 'value' => $value]) {
 				if (isset($value['shareWith'])) {
 					$id = $value['shareWith'];
 					$users[$id] = $label;
 				}
 			}
-
-			$user = $this->userSession->getUser();
```

Comment:
Since you no longer use `userSession`, could you also remove it from the constructor @luka-nextcloud?